kernel/src/engine/arrow_footer_skipping.rs

```diff
@@ -0,0 +1,359 @@
+//! An implementation of parquet row group skipping using data skipping predicates.
+use crate::expressions::{BinaryOperator, Expression, Scalar, UnaryOperator, VariadicOperator};
+use crate::schema::{DataType, PrimitiveType};
+use parquet::arrow::arrow_reader::ArrowReaderBuilder;
+use parquet::file::metadata::RowGroupMetaData;
+use parquet::file::statistics::Statistics;
+use parquet::schema::types::{ColumnDescPtr, ColumnPath};
+use std::cmp::Ordering;
+use std::collections::{HashMap, HashSet};
+
+pub fn filter_row_groups<T>(
+    reader: ArrowReaderBuilder<T>,
+    filter: &Expression,
+) -> ArrowReaderBuilder<T> {
+    let indices = reader
+        .metadata()
+        .row_groups()
+        .iter()
+        .enumerate()
+        .filter_map(|(index, row_group)| {
+            // We can only skip a row group if the filter is false (true/null means keep)
+            let keep = !matches!(RowGroupFilter::apply(filter, row_group), Some(false));
+            keep.then_some(index)
```

Comment:
Right, my bad missing the `None` case (which yours also misses btw @hntd187)

```suggestion
            RowGroupFilter::apply(filter, row_group).unwrap_or(true).then_some(index)
```

Reply:
Ryan noted that Some(true) and None produced the same results so I assumed it would be okay basically there is only one success case Some(false), unless I misunderstood?